File: packages/expo-updates/ios/EXUpdates/SelectionPolicy/EXUpdatesLauncherSelectionPolicySingleUpdate.m
//  Copyright © 2021 650 Industries. All rights reserved.

#import <EXUpdates/EXUpdatesLauncherSelectionPolicySingleUpdate.h>

#if __has_include(<EXUpdates/EXUpdates-Swift.h>)
#import <EXUpdates/EXUpdates-Swift.h>
#else
#import "EXUpdates-Swift.h"
#endif

NS_ASSUME_NONNULL_BEGIN

@interface EXUpdatesLauncherSelectionPolicySingleUpdate ()

@property (nonatomic, strong) NSUUID *updateId;

@end

@implementation EXUpdatesLauncherSelectionPolicySingleUpdate

- (instancetype)initWithUpdateID:(NSUUID *)updateId
{
  if (self = [super init]) {
    _updateId = updateId;
  }
  return self;
}

- (nullable EXUpdatesUpdate *)launchableUpdateFromUpdates:(NSArray<EXUpdatesUpdate *> *)updates filters:(nullable NSDictionary *)filters
{
  for (EXUpdatesUpdate *update in updates) {
    if ([update.updateId isEqual:_updateId]) {
      return update;
    }
  }
  return nil;
}

@end

NS_ASSUME_NONNULL_END

File: packages/expo-updates/ios/EXUpdates/SelectionPolicy/EXUpdatesLauncherSelectionPolicySingleUpdate.swift
//  Copyright © 2019 650 Industries. All rights reserved.

import Foundation

/**
 * A trivial EXUpdatesLauncherSelectionPolicy that will choose a single predetermined update to launch.
 */
@objcMembers
public final class EXUpdatesLauncherSelectionPolicySingleUpdate: NSObject, EXUpdatesLauncherSelectionPolicy {
  let updateId: UUID

  public required init(updateId: UUID) {
    self.updateId = updateId
  }

  public func launchableUpdate(fromUpdates updates: [EXUpdatesUpdate], filters: [String: Any]?) -> EXUpdatesUpdate? {
    return updates.first { update in
      return update.updateId == self.updateId
    }
  }
}
